test/Microsoft.ML.Tests/TrainerEstimators/TrainerEstimators.cs

```diff
@@ -71,8 +71,11 @@ public void KMeansEstimator()
 
 
             // Pipeline.
-            var pipeline = new KMeansPlusPlusTrainer(Env, featureColumn, weights: weights,
-                            advancedSettings: s => { s.InitAlgorithm = KMeansPlusPlusTrainer.InitAlgorithm.KMeansParallel; });
+            var pipeline = new KMeansPlusPlusTrainer(Env, new KMeansPlusPlusTrainer.Options {
```

Comment:
>KMeansPlusPlusTrainer [](start = 31, length = 21)

surprised... is this invoking the constructor ?

Reply:
nothing wrong with it.. the extensions return the same object too, an IEstimator. 

---
In reply to: [248377514](https://github.com/dotnet/machinelearning/pull/2158#discussion_r248377514) [](ancestors = 248377514)